cassovary-core/src/main/scala/com/twitter/cassovary/algorithms/shortestpath/ShortestPath.scala

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2015 Twitter, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this
+ * file except in compliance with the License. You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package com.twitter.cassovary.algorithms.shortestpath
+
+import com.twitter.cassovary.graph.{Node, DirectedGraph}
+
+/**
+ * A base trait for all ShortestPath algorithm implementations
+ */
+trait ShortestPath {
```

Comment:
it really is now single source shortest path, so I'd merge this into the class (also see below).